app/.server/get-git-feed.ts

```diff
@@ -0,0 +1,69 @@
+import fs from 'node:fs';
+import { z } from 'zod';
+
+const GithubCommitsAPISchema = z.array(
+  z.object({
+    sha: z.string().min(1),
+    html_url: z.string().url().min(1),
+    commit: z.object({
+      message: z.string().min(1),
+      committer: z.object({
+        date: z.string().datetime().min(1),
+      }),
+    }),
+  }),
+);
+
+const StoredCommitsSchema = z.array(
+  z.object({
+    id: z.string().min(1),
+    link: z.string().url().min(1),
+    content: z.string().min(1),
+    timestamp: z.string().datetime().min(1),
+  }),
+);
+
+const GITHUB_URL =
+  'https://api.github.com/repos/sorokya/reoserv/commits?sha=master&per_page=20';
+const DATA_FILE_PATH = 'git-feed.json';
+const MAX_FILE_AGE = 5 * 60 * 1000; // 5 minutes in milliseconds
+
+async function getGitFeed() {
+  // Check file age or existence
+  const fileStats =
+    fs.existsSync(DATA_FILE_PATH) && fs.statSync(DATA_FILE_PATH);
+  const fileAge = fileStats
+    ? Date.now() - fileStats.mtime.getTime()
+    : Number.POSITIVE_INFINITY;
+
+  if (!fileStats || fileAge > MAX_FILE_AGE) {
+    const commits = await fetchGitFeed();
+    const json = JSON.stringify(commits);
+    fs.writeFileSync(DATA_FILE_PATH, json);
+    return commits;
+  }
+
+  const fileContents = fs.readFileSync(DATA_FILE_PATH, { encoding: 'utf8' });
+  const json = JSON.parse(fileContents);
```

Comment:
Shouldn't fileContents be json and this variable be named something else?

Reply:
I'd leave it as it is. `fs.readFileSync` outputs a string, which we parse as JSON in the second step.